src/components/Box/Box.jsx

```diff
@@ -0,0 +1,405 @@
+import React, { useRef, forwardRef } from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import useMergeRefs from "hooks/useMergeRefs";
+import styles from "./Box.module.scss";
+import {
+  DISABLED,
+  BORDER,
+  BORDER_TOP,
+  BORDER_END,
+  BORDER_BOTTOM,
+  BORDER_START,
+  BORDER_COLOR,
+  ROUNDED,
+  ROUNDED_TOP,
+  ROUNDED_END,
+  ROUNDED_BOTTOM,
+  ROUNDED_START,
+  SHADOW,
+  MARGIN,
+  MARGIN_X,
+  MARGIN_Y,
+  MARGIN_TOP,
+  MARGIN_END,
+  MARGIN_BOTTOM,
+  MARGIN_START,
+  PADDING,
+  PADDING_X,
+  PADDING_Y,
+  PADDING_TOP,
+  PADDING_END,
+  PADDING_BOTTOM,
+  PADDING_START,
+  BACKGROUND_COLORS,
+  COLORS
+} from "./BoxConstants";
+
+const Box = forwardRef(
+  (
+    {
+      className,
+      id,
+      children,
+      disabled,
+      border,
+      borderTop,
+      borderEnd,
+      borderBottom,
+      borderStart,
+      borderColor,
+      rounded,
+      roundedTop,
+      roundedEnd,
+      roundedBottom,
+      roundedStart,
+      shadow,
+      margin,
+      marginX,
+      marginY,
+      marginTop,
+      marginEnd,
+      marginBottom,
+      marginStart,
+      padding,
+      paddingX,
+      paddingY,
+      paddingTop,
+      paddingEnd,
+      paddingBottom,
+      paddingStart,
+      textColor,
+      backgroundColor
+    },
+    ref
+  ) => {
+    const componentRef = useRef(null);
+    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+
+    return (
+      <div
+        ref={mergedRef}
+        className={cx(
+          styles.box,
+          className,
+          { [DISABLED.DISABLED]: disabled },
+          border,
+          borderTop,
+          borderEnd,
+          borderBottom,
+          borderStart,
+          borderColor,
+          rounded,
+          roundedTop,
+          roundedEnd,
+          roundedBottom,
+          roundedStart,
+          shadow,
+          margin,
+          marginX,
```

Comment:
what @3dyonic and @orrgottlieb  are you decide regarding marginx, marginy, paddingx, paddingy in the end?

Reply:
there are in.